rust/agama-cli/src/profile.rs

```diff
@@ -76,123 +69,256 @@ pub enum ProfileCommands {
+    }
+}
+
+/// Validate a JSON profile, by doing a HTTP client request.
+async fn validate_client(
+    client: &BaseHTTPClient,
+    url_or_path: CliInput,
+) -> anyhow::Result<ValidationOutcome> {
+    let mut url = Url::parse(&client.base_url)?;
+    // unwrap OK: only fails for cannot_be_a_base URLs like data: and mailto:
+    url.path_segments_mut()
+        .unwrap()
+        .push("profile")
+        .push("validate");
+    url_or_path.add_query(&mut url)?;
+
+    let body = url_or_path.body_for_web()?;
+    // we use plain text .body instead of .json
+    let response: Result<reqwest::Response, agama_lib::error::ServiceError> = client
+        .client
+        .request(reqwest::Method::POST, url)
+        .body(body)
+        .send()
+        .await
+        .map_err(|e| e.into());
+
+    let result = client.deserialize_or_error(response?).await;
+    result.map_err(|e| e.into())
+}
+
+async fn validate(client: &BaseHTTPClient, url_or_path: CliInput) -> anyhow::Result<()> {
+    let validity = validate_client(client, url_or_path).await?;
+    match validity {
+        ValidationOutcome::Valid => {
+            println!("{} {}", style("\u{2713}").bold().green(), validity);
+        }
+        ValidationOutcome::NotValid(_) => {
+            println!("{} {}", style("\u{2717}").bold().red(), validity);
         }
     }
     Ok(())
 }
 
-fn evaluate(path: &Path) -> anyhow::Result<()> {
-    let evaluator = ProfileEvaluator {};
-    evaluator
-        .evaluate(path, stdout())
-        .context("Could not evaluate the profile".to_string())?;
+/// Evaluate a Jsonnet profile, by doing a HTTP client request.
+/// Return well-formed Agama JSON on success.
+async fn evaluate_client(client: &BaseHTTPClient, url_or_path: CliInput) -> anyhow::Result<String> {
+    let mut url = Url::parse(&client.base_url)?;
+    // unwrap OK: only fails for cannot_be_a_base URLs like data: and mailto:
+    url.path_segments_mut()
+        .unwrap()
+        .push("profile")
+        .push("evaluate");
+    url_or_path.add_query(&mut url)?;
+
+    let body = url_or_path.body_for_web()?;
+    // we use plain text .body instead of .json
+    let response: Result<reqwest::Response, agama_lib::error::ServiceError> = client
+        .client
+        .request(reqwest::Method::POST, url)
+        .body(body)
+        .send()
+        .await
+        .map_err(|e| e.into());
+
+    let output: Box<serde_json::value::RawValue> = client.deserialize_or_error(response?).await?;
+    Ok(output.to_string())
+}
+
+async fn evaluate(client: &BaseHTTPClient, url_or_path: CliInput) -> anyhow::Result<()> {
+    let output = evaluate_client(client, url_or_path).await?;
+    println!("{}", output);
     Ok(())
 }
 
-async fn import(url_string: String, dir: Option<PathBuf>) -> anyhow::Result<()> {
+async fn import(client: BaseHTTPClient, url_string: String) -> anyhow::Result<()> {
+    // useful for store_settings
     tokio::spawn(async move {
         show_progress().await.unwrap();
```

Comment:
This code will not work anymore because the D-Bus server is not accessible from here. We might need to rely on the HTTP API instead.

Reply:
Is that a recent change? Which PR?

Reply:
Duh, you mean when working with the `--api` option, right

Reply:
Yes, I meant from a remote system (`--api`). You cannot connect to the D-Bus server of the system you are installing.